### PR TITLE
Update changelog.md

### PR DESCRIPTION
**Pull request by jp4a50**
2023-07-07 release notes.

### CODE REVIEW

File: content/workers/platform/changelog.md [hunk 1]
```diff
@@ -8,6 +8,11 @@ rss: file
 
 # Changelog
 
+## 2023-07-07
+
+- An implementation of process.env is now available when using the `nodejs_compat` compatibility flag.
+- An implementation of the Node.js diagnostics_channel API is now available when using the `nodejs_compat` compatibility flag.
```
Comment (irvinebroque) [inline content/workers/platform/changelog.md]: Going to handle comments on https://github.com/cloudflare/cloudflare-docs/pull/9571 and get that merged, then update this to link to there.

File: content/workers/platform/changelog.md [hunk 2]
```diff
@@ -8,6 +8,11 @@ rss: file
 
 # Changelog
 
+## 2023-07-07
+
+- An implementation of process.env is now available when using the `nodejs_compat` compatibility flag.
```
Comment (irvinebroque) [inline content/workers/platform/changelog.md]:
```suggestion
- An implementation of the [`process.env`](/workers/runtime-apis/nodejs/process) API from Node.js is now available when using the `nodejs_compat` compatibility flag.
```

File: content/workers/platform/changelog.md (hunk 1 above)
Comment (irvinebroque) [inline content/workers/platform/changelog.md]:
```suggestion
- An implementation of the [`diagnostics_channel`](/workers/runtime-apis/nodejs/diagnostics-channel) API from Node.js is now available when using the `nodejs_compat` compatibility flag.
```

Comment (jp4a50) [reply content/workers/platform/changelog.md]: Sounds good. I was wondering whether there were any docs for it.

**Review by irvinebroque — APPROVED** [review]

**Pull request merged**